### PR TITLE
DOCSP-40133: custom auth provider

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kafka-connector/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-40133>
Staging - https://preview-mongodbrustagir.gatsbyjs.io/kafka-connector/DOCSP-40133-custom-authprovider/security-and-authentication/custom-auth/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST? **errors are unrelated and resolved in #162**
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

**Review by mcmorisi — CHANGES_REQUESTED** [review]

**Comment by RWaltersMA** [comment]
Its a custom authentication provider, not specific to AWS IAM. We are providing a sample to do AWS IAM Auth though, the way the docs are written sounds like it is just AWS IAM support.

**Review by mcmorisi — APPROVED** [review]
LGTM. Really great job on this!

File: source/security-and-authentication/custom-auth.txt [hunk 1]
```diff
@@ -72,13 +72,13 @@ by using AWS IAM authentication:
...
       }
    }
 
 In this example, the ``roleArn`` value is the IAM Role of the user group that has
-access to MongoDB Atlas. In the AWS IAM console, we gave ``AssumeRole``
-permissions on the IAM account that is running {+kafka-connect+} to this
-Atlas Group IAM Role.
+access to MongoDB Atlas. In the AWS IAM console, the IAM account that is
+running {+kafka-connect+} has ``AssumeRole`` permissions to this Atlas
+Group IAM Role.
```
Comment (RWaltersMA) [inline source/security-and-authentication/custom-auth.txt:84]: Atlas User Group

File: source/security-and-authentication/custom-auth.txt [hunk 2]
```diff
@@ -72,13 +72,13 @@ by using AWS IAM authentication:
         "value.converter.schemas.enable": false,
         "database": "<db>",
         "collection": "<collection>",
-        "mongo.custom.auth.mechanism.enable":"true",
-        "mongo.custom.auth.mechanism.providerClass":"com.mongodb.SampleAssumeRoleCredential",
-        "mongodbaws.auth.mechanism.roleArn":"arn:aws:iam::99999999:role/KafkaAtlasRole"
+        "mongo.custom.auth.mechanism.enable": "true",
+        "mongo.custom.auth.mechanism.providerClass": "com.mongodb.SampleAssumeRoleCredential",
+        "mongodbaws.auth.mechanism.roleArn": "arn:aws:iam::99999999:role/KafkaAtlasRole"
```
Comment (RWaltersMA) [inline source/security-and-authentication/custom-auth.txt:77]: replace arn:aws:iam with <ARN of AWS IAM Role>

Comment (RWaltersMA) [reply source/security-and-authentication/custom-auth.txt:77]: aws iam ARN something like that

**Comment by RWaltersMA** [comment]
LGTM

**Pull request merged**